errtracker/errtracker.go

```diff
@@ -160,6 +168,104 @@ func detectVirt() string {
 	return strings.TrimSpace(string(output))
 }
 
+func journalError() string {
+	// TODO: look into using systemd package (needs refactor)
+	output, err := exec.Command("journalctl", "--utc", "--boot", "--priority=warning..err", "--lines=1000").CombinedOutput()
+	if err != nil {
+		if len(output) == 0 {
+			return fmt.Sprintf("error: %v", err)
+		}
+		output = append(output, fmt.Sprintf("\nerror: %v", err)...)
+	}
+	return string(output)
+}
+
+func procCpuinfoMinimal() string {
+	buf, err := ioutil.ReadFile(procCpuinfo)
+	if err != nil {
+		// if we can't read cpuinfo, we want to know _why_
+		return fmt.Sprintf("error: %v", err)
+	}
+	idx := bytes.LastIndex(buf, []byte("\nprocessor\t:"))
+	// if not found, return whole buffer; otherwise, return from just after the \n
+	return string(buf[idx+1:])
+}
+
+func procExe() string {
+	out, err := os.Readlink(procSelfExe)
+	if err != nil {
+		return fmt.Sprintf("error: %v", err)
+	}
+	return out
+}
+
+func procCwd() string {
+	out, err := os.Readlink(procSelfCwd)
+	if err != nil {
+		return fmt.Sprintf("error: %v", err)
+	}
+	return out
+}
+
+func procCmdline() string {
+	out, err := ioutil.ReadFile(procSelfCmdline)
+	if err != nil {
+		return fmt.Sprintf("error: %v", err)
+	}
+	return string(out)
+}
+
+func environ() string {
+	safeVars := []string{
+		"SHELL", "TERM", "LANGUAGE", "LANG", "LC_CTYPE",
+		"LC_COLLATE", "LC_TIME", "LC_NUMERIC",
+		"LC_MONETARY", "LC_MESSAGES", "LC_PAPER",
+		"LC_NAME", "LC_ADDRESS", "LC_TELEPHONE",
+		"LC_MEASUREMENT", "LC_IDENTIFICATION", "LOCPATH",
+	}
+	out := make([]string, 0, len(safeVars)+4)
+	for _, env := range osEnviron() {
+		idx := strings.IndexByte(env, '=')
+		if idx < 4 {
+			continue
+		}
+		if len(env) == idx+1 {
+			continue
+		}
+
+		switch k := env[:idx]; k {
+		case "PATH":
```

Comment:
this mirrors what apport does, but really I think a more useful approach would be to report PATH, with the known entries passed through as is, and custom ones blanked out. I.e. something like,
if `PATH` is `/home/ubuntu/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/opt/stuff/bin:/snap/bin`, then the the report has `PATH=/home/john/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin`.

WDYT?

Reply:
Looks better. If I read this correctly, the current code will produce eg: `PATH=(custom,user,no snap)`, this does not look very useful. One could argue that it's not sending the 'standard' PATH, but I can't believe it's really that much data to handle.